spec/03-plugins/17-jwt/03-access_spec.lua

```diff
@@ -277,6 +281,59 @@ describe("Plugin: jwt (access)", function()
       })
       assert.res_status(200, res)
     end)
+    it("finds the JWT if given in cookie crumble", function()
+      PAYLOAD.iss = jwt_secret.key
+      local jwt = jwt_encoder.encode(PAYLOAD, jwt_secret.secret)
+      local res = assert(proxy_client:send {
+        method  = "GET",
+        path    = "/request",
+        headers = {
+          ["Host"] = "jwt9.com",
+          ["Cookie"] = "crumble=" .. jwt .. "; path=/;domain=.jwt9.com"
+        }
+      })
+      assert.res_status(200, res)
+    end)
+    it("finds the JWT if given in cookie silly", function()
+      PAYLOAD.iss = jwt_secret.key
+      local jwt = jwt_encoder.encode(PAYLOAD, jwt_secret.secret)
+      local res = assert(proxy_client:send {
+        method  = "GET",
+        path    = "/request",
+        headers = {
+          ["Host"] = "jwt9.com",
+          ["Cookie"] = "silly=" .. jwt .. "; path=/;domain=.jwt9.com"
+        }
+      })
+      assert.res_status(200, res)
+    end)
+    it("reports a 401 if the JWT in cookie is corrupt", function()
+      PAYLOAD.iss = jwt_secret.key
+      local jwt = 'no-way-this-works' .. jwt_encoder.encode(PAYLOAD, jwt_secret.secret)
+      local res = assert(proxy_client:send {
+        method  = "GET",
+        path    = "/request",
+        headers = {
+          ["Host"] = "jwt9.com",
+          ["Cookie"] = "silly=" .. jwt .. "; path=/;domain=.jwt9.com"
+        }
+      })
+      local body = assert.res_status(401, res)
+      assert.equal([[{"message":"Bad token; invalid JSON"}]], body)
+    end)
+    it("no cookie with JWT token, defaults to 'Authorization'", function()
+      PAYLOAD.iss = jwt_secret.key
+      local jwt = jwt_encoder.encode(PAYLOAD, jwt_secret.secret)
+      local res = assert(proxy_client:send {
+        method  = "GET",
+        path    = "/request",
+        headers = {
+          ["Host"] = "jwt9.com",
+          ["Authorization"] = "Bearer " .. jwt,
+        }
+      })
+      assert.res_status(200, res)
+    end)
```

Comment:
While the state of the code today implies that a fallback on the Authorization header will also imply an HTTP 403 will occur when necessary, this might not be the case in the future, say, if a refactor of the code occurs. I'd feel more comfortable if we included a test for HTTP 403 on a plugin with `cookie_names`, when not sending cookies nor an Authorization header.

What do you think?

Reply:
I believe it should be a 401  (Unauthenticated). I added the test to include this case.

Reply:
Oops, sorry - sent this review from a meeting... I think the new one you added is a nice addition, thanks. I also think we need the HTTP 403 case (Forbidden) which is present in other tests paths, but not the one from a cookie.

Reply:
I added one of the JWT cookie case.  I left it out, as all the other tests already validate the proper operation after a token was found. The added tests for cookie_names merely focused on finding the cookie token.

Reply:
Yes, those are integration tests and leaving such a test out would mean testing this behavior with expectations and assumptions made about the underlying implementation. Preferably, integration tests don't make any assumptions about the tested code and does assume that the system is a black box. Thank you for adding it.